test/features/voucher/data/datasources/voucher_remote_data_source_test.dart

```diff
@@ -0,0 +1,58 @@
+import 'package:coffeecard/core/network/network_request_executor.dart';
+import 'package:coffeecard/features/voucher/data/datasources/voucher_remote_data_source.dart';
+import 'package:coffeecard/features/voucher/data/models/redeemed_voucher_model.dart';
+import 'package:coffeecard/generated/api/coffeecard_api.swagger.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:fpdart/fpdart.dart';
+import 'package:mockito/annotations.dart';
+import 'package:mockito/mockito.dart';
+
+import 'voucher_remote_data_source_test.mocks.dart';
+
+@GenerateMocks([CoffeecardApi, NetworkRequestExecutor])
+void main() {
+  late VoucherRemoteDataSource remoteDataSource;
+  late MockCoffeecardApi apiV1;
+  late MockNetworkRequestExecutor executor;
+
+  setUp(() {
+    executor = MockNetworkRequestExecutor();
+    apiV1 = MockCoffeecardApi();
+    remoteDataSource = VoucherRemoteDataSource(
+      apiV1: apiV1,
+      executor: executor,
+    );
+  });
+
+  group('redeemVoucher', () {
+    test('should call executor', () async {
```

Comment:
test name and assertion step differ ("Should call executor" would check if executor was called in the assertion step)